backend/Squirrel.Core/Squirrel.Core.BLL/Services/UserService.cs

```diff
@@ -0,0 +1,86 @@
+﻿using AutoMapper;
+using Microsoft.EntityFrameworkCore;
+using Squirrel.Core.BLL.Extensions;
+using Squirrel.Core.BLL.Interfaces;
+using Squirrel.Core.BLL.Services.Abstract;
+using Squirrel.Core.Common.DTO.Auth;
+using Squirrel.Core.Common.Security;
+using Squirrel.Core.DAL.Context;
+using Squirrel.Core.DAL.Entities;
+using Squirrel.Shared.Exceptions;
+
+namespace Squirrel.Core.BLL.Services;
+
+public sealed class UserService : BaseService, IUserService
+{
+    private const int MaxNameLength = 25;
+    private const int MinNameLength = 2;
+    
+    public UserService(SquirrelCoreContext context, IMapper mapper) : base(context, mapper)
+    {
+    }
+
+    public async Task<User?> GetUserByEmailAsync(string email)
+        => await _context.Users.FirstOrDefaultAsync(u => u.Email == email);
+
+    public async Task<User?> GetUserByUsernameAsync(string username)
+        => await _context.Users.FirstOrDefaultAsync(u => u.Username == username);
+
+    public async Task<User> CreateUserAsync(UserRegisterDto userDto, bool isGoogleAuth)
+    {
+        if (await GetUserByUsernameAsync(userDto.Username) is not null)
+        {
+            if (isGoogleAuth)
+            {
+                // Google registration must not fail because of the same username.
+                userDto.Username = GenerateRandomUsername();
```

Comment:
We will show userName, so please generate it based on firstname + lastname

Reply:
i do this in mapping, but when two users have the same firstname + lastname I generate random unique one

Reply:
and what if this duplicating firstname + lastname takes exactly 25 characters, so we cannot add any random characters to it?)

Reply:
in this case, if we use 5 random characters, we can remove the last 5 characters of the lastname and add 5 random characters. we only want it to be in a more user-friendly format.